modelon/impact/client/experiment_definition.py

```diff
@@ -544,6 +518,13 @@ def with_modifiers(self, modifiers: dict = None):
                 custom_function).with_modifiers({'inertia1.J': Choices(0.1, 0.9),
                 'inertia2.J': Range(0.1, 0.5, 3)})
         """
+        if not isinstance(expansion, BaseExpansion):
+            raise TypeError(
+                "The expansion argument must be an instance of "
+                "modelon.impact.client.expansion.FullFactorial, "
+                "modelon.impact.client.expansion.Sobol or "
+                "modelon.impact.client.expansion.LatinHypercube"
```

Comment:
I think this error message could be something like:
`The expansion argument is of type '{type(expansion)}' which is not a subtype of 'ExpansionAlgorithm'` which would be more general and also indicates that you can implement your own expansion algorithm if the one you want is missing.